### PR TITLE
UPDATE: Ligand Complex models and Structure Similarity Search

### PR DESCRIPTION
**Pull request by prossotto**
build for Endogenous ligand complex models
Structure similarity search implementation
update of the Endogenous Browser to have a link to the Physiological ligands

### CODE REVIEW

File: build/management/commands/build_rfaa_models.py [hunk 1]
```diff
@@ -0,0 +1,1013 @@
...
+            struct.refined = False
+            struct.stats_text = None
+
+            #Resolution and
+            # save structure before adding M2M relations
+            struct.save()
+
+            print('BEFORE CREATING LIGAND')
+
+            # Create LigandPeptideStructure object
```
Comment (pszgaspar) [inline build/management/commands/build_rfaa_models.py:904]: these are not peptide ligands so I guess this shouldn't be here? If you want to include small molecule ligand objects it should be with StructureLigandInteraction

File: build/management/commands/build_rfaa_models.py [hunk 2]
```diff
@@ -0,0 +1,1013 @@
...
+                else:
+                    print(f"Found existing LigandPeptideStructure for structure {struct.pdb_code.index} and ligand {ligand.name}")
+            
+            except Ligand.DoesNotExist:
+                print(f"Error: Ligand with id {sd['ligand_id']} does not exist in the database.")
+            except Exception as e:
+                print(f"Error creating LigandPeptideStructure: {str(e)}")
+
+####################################################
+            Rotamer.objects.filter(structure=struct, pdbdata=pdbdata).delete()
```
Comment (pszgaspar) [inline build/management/commands/build_rfaa_models.py:928]: I'm not sure what this is here. I don't think you have built the rotamer objects but even if you did why delete them?

File: build/management/commands/build_rfaa_models.py [hunk 3]
```diff
@@ -0,0 +1,1013 @@
...
+            # Residue.objects.filter(protein_conformation=struct.protein_conformation).delete()
+
+####################################################
+            d = {}
+
+            # try:
+            #     current = time.time()
+            #     self.create_rotamers(struct, pdb_path, d)
```
Comment (pszgaspar) [inline build/management/commands/build_rfaa_models.py:936]: create_rotamers is commented, I guess you don't build residue objects then. I think we would only need them for the interactions which we are not doing now, so you could remove this and actually the complete create_rotamers function

File: build/management/commands/build_rfaa_models.py [hunk 4]
```diff
@@ -0,0 +1,1013 @@
...
+            #     self.logger.error('Error with rotamers for {}'.format(sd['pdb']))
+            #     self.rotamer_errors.append(struct)
+
+            try:
+                struct.protein_conformation.generate_sites()
+            except:
+                pass
+
+            ##### SIGNPROT
+            # beta_gamma = sd['beta_gamma']
```
Comment (pszgaspar) [inline build/management/commands/build_rfaa_models.py:955]: no signprots are in the models, remove these please

File: build/management/commands/build_rfaa_models.py [hunk 5]
```diff
@@ -0,0 +1,1013 @@
...
+                        #     res_obj = Residue.objects.get(protein_conformation__protein=beta_protconf.protein, sequence_number=res.get_id()[1])
+                        # elif chain.get_id()=='D':
+                        #     res_obj = Residue.objects.get(protein_conformation__protein=gamma_protconf.protein, sequence_number=res.get_id()[1])
+                        r = StructureModelpLDDT(structure=struct, residue=res_obj, pLDDT=plddt)
+                        resis.append(r)
+                    except Residue.DoesNotExist:
+                        continue
+            StructureModelpLDDT.objects.bulk_create(resis)
+
+            # try:
```
Comment (pszgaspar) [inline build/management/commands/build_rfaa_models.py:1003]: please clean this part up as well, we'll add compute_interactions and contact network when we need it

File: structure/assign_gn_foldseek.py [hunk 6]
```diff
@@ -0,0 +1,394 @@
+from django.conf import settings
```
Comment (pszgaspar) [inline structure/assign_gn_foldseek.py:1]: I would have preferred if you modified the assign_gn script to incorporate the foldseek gn assign. This is a duplicate file in a way. Can we integrate it into that script? What is unique about this script compared to the original? If nothing else, how about inheriting the class and only overwriting the functions that are different?

File: structure/functions.py [hunk 7]
```diff
@@ -1169,31 +1169,162 @@ def parse_ligands(self):
                 print('Cannot find information for complex {}'.format(complex))
 
 
+# class ParseAFComplexModels():
```
Comment (pszgaspar) [inline structure/functions.py:1172]: please remove this unused commented function

File: structure/templates/ligand_complex_details.html [hunk 8]
```diff
@@ -0,0 +1,1047 @@
+{% extends "home/base.html" %}
```
Comment (pszgaspar) [inline structure/templates/ligand_complex_details.html:1]: please remove all the commented stuff, we'll add functionality when needed

File: structure/templates/ligand_complex_models.html [hunk 9]
```diff
@@ -0,0 +1,569 @@
...
+                     <td> {{ model.publication_date }} </td>
+
+                </tr>
+            {% endfor %}
+            </tbody>
+        </table>
+    </div>
+</div>
+<div id="superposition-modal" class="modal">
+  <!-- Modal content -->
```
Comment (pszgaspar) [inline structure/templates/ligand_complex_models.html:544]: this we can keep here commented as I am pretty sure we will need to add this functionality in the near future

**Review by pszgaspar — CHANGES_REQUESTED** [review]
I made some comments, mostly cleanup related but there are a couple other things. Please go through them and we can zoom about the bigger changes.

File: structure/assign_gn_foldseek.py (hunk 6 above)
Comment (prossotto) [reply structure/assign_gn_foldseek.py:1]: Yes that shouldn't be there, I was just testing something, I will take it out.

**Pull request merged**